scripts/presigned-s3-urls.sh

```diff
@@ -1,29 +1,39 @@
 #!/bin/bash
 
 # Set the usage message
-usage="Usage: $0 <dir_path> <s3_uri> <output_file>"
+usage="Usage: $0 <s3_uri> <output_file> [<expires_in>]"
 
 # Example invocation
-# from ipa/input_data_S02/
-# ../scripts/presigned-s3-urls.sh encryptions/1B_cat/30_shards/ s3://stg-ipa-encrypted-reports/testing-sharded-data/1B/30_shards presigned_urls_30_shards.txt
+# ../scripts/presigned-s3-urls.sh s3://stg-ipa-encrypted-reports/testing-sharded-data/1B/30_shards presigned_urls_30_shards.txt
 
 # Check if the correct number of arguments were provided
-if [ $# -ne 3 ]; then
+if [ $# -lt 2 ] || [ $# -gt 3 ]; then
   echo "$usage"
   exit 1
 fi
 
-# Set the directory path and S3 URI from the command-line arguments
-dir_path="$1"
-s3_uri="$2"
-output_file="$3"
+# Set and validate the S3 URI, output_file, and expires_in from the command-line arguments
+s3_uri="$1"
+output_file="$2"
+# Check if the output file already exists
+if [ -f "$output_file" ]; then
+  echo "Error: Output file '$output_file' already exists. Please remove it before running this script."
+  exit 1
+fi
+
+# default expires_in: 7 days (7 * 24 * 60 * 60) - 1. this is the max allowed
+expires_in="${3:-604799}"
```

Comment:
nit - can we use hours/days here? I don't think there is a reason for us to use seconds

Reply:
original intention was to keep it as close to the aws api as possible, but we can do whatever. preference for hours vs days?